torchtext/csrc/vectors.cpp

```diff
@@ -0,0 +1,68 @@
+#include <string>
+#include <torch/script.h>
+
+namespace torchtext {
+namespace {
+
+struct Vectors : torch::CustomClassHolder {
+private:
+  std::unordered_map<std::string, torch::Tensor> stovectors_;
+
+public:
+  // tokens_, vectors_, and unk_tensor_ holds the serialized params passed in
+  // during initialization. We need this because we need to be able to serialize
+  // the model so that we can save the scripted object. Pickle will get the
+  // serialized model from these members, thus they needs to be public.
+  std::vector<std::string> tokens_;
+  std::vector<torch::Tensor> vectors_;
+  torch::Tensor unk_tensor_;
+
+  explicit Vectors(std::vector<std::string> tokens,
+                   std::vector<torch::Tensor> vectors, torch::Tensor unk_tensor)
+      : tokens_(tokens), vectors_(vectors), unk_tensor_(unk_tensor) {
+    for (std::size_t i = 0; i < tokens.size(); i++) {
+      // guarding against size mismatch of vectors and tokens
```

Comment:
@zhangguanheng66 @cpuhrsch lmk what you think of this part. Without adding this check, I'd get a segfault during the instantiation of the cpp `Vectors` class.